src/pubsub.ts

```diff
@@ -677,6 +759,39 @@ export class PubSub {
     }
   }
 
+  /**
+   * Get a list of schemas associated with your project.
+   *
+   * The returned AsyncIterable will resolve to {@link Schema} objects.
+   * These can be used for further interactions with each schema.
+   *
+   * This method returns an async iterable. These objects can be adapted
+   * to work in a Promise/then framework, as well as with callbacks, but
+   * this discussion is considered out of scope for these docs.
+   *
+   * @see [Schemas: list API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.schemas/list}
+   * @see [More about async iterators]{@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of}
+   *
+   * @param {object} [options] Request configuration options, outlined
+   *   here: https://googleapis.github.io/gax-nodejs/interfaces/CallOptions.html.
+   * @returns {AsyncIterable<Schema>}
+   *
+   * @example
+   * for await (const s of pubsub.listSchemas()) {
+   *   const moreInfo = await s.get();
+   * }
+   */
+  async *listSchemas(options?: CallOptions): AsyncIterable<Schema> {
+    const client = await this.getSchemaClient_();
+    const query = {
+      parent: this.name,
+      view: google.pubsub.v1.SchemaView.BASIC,
```

Comment:
Is there a way for users to pass in a different schema view here? We need to provide both Basic/Full views as options to be returned by `list`.